agents/ahv/fence_ahv.py

```diff
@@ -0,0 +1,744 @@
+                        raise AHVFenceAgentException from err
+
+                return resp
+
+        def _power_cycle_vm(self, vm_uuid):
+                """
+                Power cycle a VM
+
+                Args:
+                vm_uuid(str): VM UUID string
+
+                Returns:
+                object: HTTP response object.
+                """
+                if not vm_uuid:
+                        logging.error("VM UUID was not provided")
+                        raise AHVFenceAgentException("VM UUID not provided")
+
+                resp = None
+                vm_url = self.vm_url + f"/{vm_uuid}/$actions/power-cycle"
+                logging.info("Sending request to power cycle VM, %s", vm_url)
+
+                try:
+                        header_str = self._get_headers(vm_uuid)
+                        resp = self.request(url=vm_url, method='POST',
+                                            headers=header_str, verify=self.verify)
+                except NutanixClientException as err:
+                        logging.error("Failed to power on VM %s", vm_uuid)
+                        raise AHVFenceAgentException from err
+
+                return resp
+
+        def _wait_for_task(self, task_uuid, timeout=None):
+                """
+                Wait for task completion
+
+                Args:
+                    task_uuid(str): Task UUID
+                    timeout(int): Timeout value to wait before failing
+
+                Returns:
+                    None
+                """
+                if not task_uuid:
+                        logging.error("Task UUID was not provided")
+                        raise AHVFenceAgentException("Task UUID not provided")
+
+                task_url = f"{self.task_url}/{task_uuid}"
+                header_str = self._get_headers()
+                task_resp = None
+                interval = 10
+                task_status = None
+
+                if not timeout:
+                        timeout = MIN_TIMEOUT
+
+                while task_status != 'SUCCEEDED':
+                        if task_status == 'FAILED':
+                                raise NutanixClientException(f"Task failed, task uuid: {task_uuid}")
+
+                        try:
+                                task_resp = self.request(url=task_url, method='GET',
+                                                         headers=header_str, verify=self.verify)
+                                task_status = task_resp.json()['data']['status']
+                        except NutanixClientException as err:
+                                logging.error("Unable to retrieve task status")
+                                raise AHVFenceAgentException from err
+
+                        if task_status == 'SUCCEEDED':
+                                break
+
+                        time.sleep(interval)
+                        timeout = timeout - interval
+
+                        if task_status == 'SUCCEEDED':
```

Comment:
Why do we have this line duplicated? this is the same as line 349!

Reply:
Thanks, I missed it. I think I might have mistakenly copied twice.